classes/requests/helpers/class-paysoncheckout-for-woocommerce-helper-order.php

```diff
@@ -105,6 +111,22 @@ public function get_product_tax_rate( $order, $order_item ) {
 		return 0;
 	}
 
+	/**
+	 * Get the product sku or id.
+	 *
+	 * @param object $product The WooCommerce Product.
+	 * @return string
+	 */
+	public function get_product_sku( $product ) {
+		if ( $product->get_sku() ) {
+			$item_reference = $product->get_sku();
+		} else {
+			$item_reference = $product->get_id();
+		}
```

Comment:
This will only work as long as the product has not been removed from the store. IF that is the case then this will break and throw an error due to `$item_reference` not being defined on return. Also in these cases `$product` will be `false`, and neither `false->get_sku()` or `false->get_id()` are methods that exist.

This is not critical though, since stores should not be deleting products before all orders are processed that contains them, but its still a potential issue that we get support tickets for. In other plugins we have solved this by a if check on the `$product` variable and returning the `$order_item` name as the reference instead. We have also in some cases provided a filter so customers can modify it on missing products based on their need.


For example:
```
function get_product_reference(WC_Order_Item_Product $order_item) {
    $reference = $order_item->get_name();
    $product = $order_item['variation_id'] ? wc_get_product( $order_item['variation_id'] ) : wc_get_product( $order_item['product_id'] );
    if($product) {
        if ( $product->get_sku() ) {
            $reference = $product->get_sku();
        } else {
            $reference = $product->get_id();
        }
    }

    return apply_filters( 'payson_order_item_reference', $reference, $order_item );
}
```

I think we might have similar issues in other places in the code as well when getting the reference from a product that could be deleted after the fact during order management.

Reply:
This is how we're currently handling the item reference when _placing_ the order, so it is basically copy-paste. Your code example is definitely an improvement, so I'll modify the code accordingly. Do you know if the suggested function will guarantee that the item reference will be the same as the one on order creation if the item is deleted? This is necessary since we check the item reference when populating order lines. I can otherwise test for myself.

@MichaelBengtsson